src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -159,60 +159,48 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 constraintClauses, body, expressionBody, semicolonToken);
         }
 
-        private ParameterListSyntax GenerateParameterList(
-            SemanticModel semanticModel, AnonymousFunctionExpressionSyntax anonymousFunction, INamedTypeSymbol delegateType, CancellationToken cancellationToken)
+        private static ParameterListSyntax GenerateParameterList(
+            SemanticModel semanticModel, AnonymousFunctionExpressionSyntax function, INamedTypeSymbol delegateType, CancellationToken cancellationToken)
         {
-            switch (anonymousFunction)
-            {
-                case SimpleLambdaExpressionSyntax simpleLambda:
-                    return GenerateSimpleLambdaParameterList(semanticModel, simpleLambda, delegateType.DelegateInvokeMethod, cancellationToken);
-                case ParenthesizedLambdaExpressionSyntax parenthesizedLambda:
-                    return GenerateParenthesizedLambdaParameterList(semanticModel, parenthesizedLambda, delegateType.DelegateInvokeMethod, cancellationToken);
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(anonymousFunction);
-            }
-        }
+            var (parameterList, simpleParameter) = GetParameterListOrParameter(function);
 
-        private ParameterListSyntax GenerateSimpleLambdaParameterList(
-            SemanticModel semanticModel, SimpleLambdaExpressionSyntax lambdaExpression, IMethodSymbol delegateInvokeMethod, CancellationToken cancellationToken)
-        {
-            var parameter = semanticModel.GetDeclaredSymbol(lambdaExpression.Parameter, cancellationToken);
-            var type = parameter?.Type.GenerateTypeSyntax() ?? s_objectType;
+            int i = 0;
+
+            return parameterList != null
+                ? parameterList.ReplaceNodes(parameterList.Parameters, (parameterNode, _) => PromoteParameter(parameterNode, i++))
+                : SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList<ParameterSyntax>().Add(PromoteParameter(simpleParameter, 0)));
```

Comment:
instead of calling .Add, just pass PromoteParameter directory to SeparatedList();

Reply:
ok... with new[]? this was there before, so I presumed other people were smarter than me and didn't change it

Reply:
> ok... with new[]?

No, .SeparatedList can take a single item and make a list out of it.

Reply:
you mean SingletoneSeparatedList?

Reply:
Sure :)